gst/edge/edge_sink.c

```diff
@@ -0,0 +1,344 @@
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+  g_object_class_install_property (gobject_class, PROP_PORT,
+      g_param_spec_uint ("port", "Port",
+          "The port to send (0=random available port)", 0, 65535, DEFAULT_PORT,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  gst_element_class_add_pad_template (gstelement_class,
+      gst_static_pad_template_get (&sinktemplate));
+
+  gst_element_class_set_static_metadata (gstelement_class,
+      "EdgeSink", "Sink/Edge",
+      "Publish incoming streams", "Samsung Electronics Co., Ltd.");
+
+  gstbasesink_class->start = gst_edgesink_start;
+  gstbasesink_class->render = gst_edgesink_render;
+  gstbasesink_class->set_caps = gst_edgesink_set_caps;
+
+  GST_DEBUG_CATEGORY_INIT (GST_CAT_DEFAULT,
+      GST_EDGE_ELEM_NAME_SINK, 0, "Edge sink");
+}
+
+/**
+ * @brief initialize the new element
+ */
+static void
+gst_edgesink_init (GstEdgeSink * self)
+{
+  self->host = g_strdup (DEFAULT_HOST);
+  self->port = DEFAULT_PORT;
+}
+
+/**
+ * @brief set property
+ */
+static void
+gst_edgesink_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  GstEdgeSink *self = GST_EDGESINK (object);
+
+  switch (prop_id) {
+    case PROP_HOST:
+      gst_edgesink_set_host (self, g_value_get_string (value));
+      break;
+    case PROP_PORT:
+      gst_edgesink_set_port (self, g_value_get_uint (value));
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief get property
+ */
+static void
+gst_edgesink_get_property (GObject * object, guint prop_id, GValue * value,
+    GParamSpec * pspec)
+{
+  GstEdgeSink *self = GST_EDGESINK (object);
+
+  switch (prop_id) {
+    case PROP_HOST:
+      g_value_set_string (value, gst_edgesink_get_host (self));
+      break;
+    case PROP_PORT:
+      g_value_set_uint (value, gst_edgesink_get_port (self));
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief finalize the object
+ */
+static void
+gst_edgesink_finalize (GObject * object)
+{
+  GstEdgeSink *self = GST_EDGESINK (object);
+  if (self->host) {
+    g_free (self->host);
+    self->host = NULL;
+  }
+
+  if (self->edge_h) {
+    nns_edge_release_handle (self->edge_h);
+    self->edge_h = NULL;
+  }
+
+  G_OBJECT_CLASS (parent_class)->finalize (object);
+}
+
+/**
+ * @brief start processing of edgesink
+ */
+static gboolean
+gst_edgesink_start (GstBaseSink * basesink)
+{
+  GstEdgeSink *self = GST_EDGESINK (basesink);
+
+  int ret;
+  char *port = NULL;
+  GstCaps *caps;
+
+  /** @todo support other connect_types */
+  nns_edge_connect_type_e connect_type = NNS_EDGE_CONNECT_TYPE_TCP;
+
+  ret =
+      nns_edge_create_handle ("TEMP_ID", connect_type,
+      NNS_EDGE_FLAG_SEND | NNS_EDGE_FLAG_SERVER, &self->edge_h);
```

Comment:
NNStreamer-edge does not yet implemented for this flag. Yoy need to update nns-edge also.